### ISSUE TITLE
Needs better documentation.

### ISSUE THREAD

**Issue by dsouzadyn** [deepmind/sonnet#27]

**Comment by dm-jrae** [comment]
What did you find unclear?

**Comment by dsouzadyn** [comment]
In the readme specifically there's a snippet showing that you get the training/test data after calling functions. I would like to know what format does this data use, is it the same as Tensorflow or something else?

**Comment by malcolmreynolds** [comment]
@dsouzadyn That's a hypothetical example (Sonnet is not a data loading library, we just provide one in the examples folder as motivation). In this case the data has to be a single TensorFlow Tensor, because it is being passed into a Linear module and you can't apply a Linear to anything but a single Tensor, but in general the data could be something different, and it's up to you to write a module that deals with your specific shape of data. Sonnet doesn't make restrictions on what the input to a module should be - it's common to have a single Tensor, but you can also write modules that take in tuples, namedtuples, etc - as long as the leaf values in these structures are either TensorFlow Tensors or python primitives (which can then be converted to Tensors) you should be fine.

Please see rnn_shakespeare.py for an example - the dataset object included there does something similar and returns a tuple of (observations, targets). You could then make a module which internally splits this apart, applies an inner model to the observations to make predictions, then a loss function, etc.

The fact that modules take TensorFlow Tensor objects as input and output is already documented in the Readme, under the question "Q: Can I mix this with other high level TF APIs, eg TF Slim?". I'm closing this but please reopen if you find that this section is insufficient.
